views.py

```diff
@@ -799,3 +812,270 @@ def typesetting_notify_proofreader(request, article_id, assignment_id):
+
+@decorators.has_journal
+@decorators.production_user_or_editor_required
+def typesetting_manage_proofing_assignment(request, article_id, assignment_id):
+    article = get_object_or_404(
+        submission_models.Article,
+        pk=article_id,
+        journal=request.journal,
+    )
+    assignment = get_object_or_404(
+        models.GalleyProofing,
+        pk=assignment_id,
+    )
+    rounds = models.TypesettingRound.objects.filter(article=article)
+    proofreaders = logic.get_proofreaders(
+        article,
+        rounds[0],
+        assignment=assignment,
+    )
+
+    form = forms.EditProofingAssignment(
+        instance=assignment,
+    )
+
+    if request.POST:
+
+        if 'action' in request.POST:
+
+            action = request.POST.get('action')
+
+            if action == 'cancel':
+                assignment.cancel()
+                messages.add_message(
+                    request,
+                    messages.SUCCESS,
+                    'Proofing task cancelled.',
+                )
+            elif action == 'reset':
+                assignment.reset()
+                messages.add_message(
+                    request,
+                    messages.SUCCESS,
+                    'Proofing task reset.',
+                )
+
+            return redirect(
+                reverse(
+                    'typesetting_article',
+                    kwargs={'article_id': article.pk},
+                )
+            )
+
+        form = forms.EditProofingAssignment(
+            request.POST,
+            instance=assignment,
+        )
+
+        if form.is_valid():
+            form.save()
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                'Assignment updated.',
+            )
+
+            return redirect(
+                reverse(
+                    'typesetting_manage_proofing_assignment',
+                    kwargs={
+                        'article_id': article.pk,
+                        'assignment_id': assignment.pk,
+                    }
+                )
+            )
+
+    template = 'typesetting/typesetting_manage_proofing_assignment.html'
+    context = {
+        'article': article,
+        'assignment': assignment,
+        'proofreaders': proofreaders,
+        'form': form,
+    }
+
+    return render(request, template, context)
+
+
+@decorators.proofreader_user_required
+def typesetting_proofreading_assignments(request):
+    assignments = models.GalleyProofing.objects.filter(
+        proofreader=request.user,
+    )
+
+    template = 'typesetting/typesetting_proofing_assignments.html'
+    context = {
+        'assignments': assignments,
+    }
+
+    return render(request, template, context)
+
+
+@decorators.proofreader_user_required
+def typesetting_proofreading_assignment(request, assignment_id):
+    assignment = get_object_or_404(
+        models.GalleyProofing,
+        pk=assignment_id,
+        proofreader=request.user,
```

Comment:
I don't think the Editor will need to access to proofreader interface like we do currently. They will have a box to add info to corrections separately from adding comments.

Reply:
> Editor

I Meant staff and superusers won't have access either

Reply:
> I Meant staff and superusers won't have access either

There are a bunch of views that staff and superusers can't access in the workflow because they have this kind of check, that said its worth just using the decorator so we can access it if needed.